tests/generate_source/build.bp

```diff
@@ -238,14 +238,26 @@ bob_generate_source {
     name: "gen_source_globbed_implicit_sources",
     implicit_srcs: ["*.implicit.src"],
     out: ["validate_globbed_implicit_dependency.c"],
-    cmd: "echo 'int main(void) { return 0; }' > ${out}",
+    tool: "join_srcs.py",
+    cmd: "python ${tool} --src-dir ${module_dir} --use-a --out ${out}",
+    build_by_default: true,
+}
+
+bob_generate_source {
+    name: "gen_source_globbed_exclude_implicit_sources",
+    implicit_srcs: ["*.src"],
+    exclude_implicit_srcs: ["an.implicit.src", "bn.src"],
```

Comment:
How does this test verify that these implicit sources have been excluded?

Reply:
In two ways:
1. If an.implicit.src will be pulled in, it will fail because resulting binary already contain main() function.
2. If bn.src will be pulled in, #error directive will cause compile to fail.

Reply:
OK, that tests that `join_srcs.py` doesn't accidentally use more than one implicit source.

But it doesn't test the `exclude_implicit_srcs` property, because `implicit_srcs` never gets written to the command line anyway! So even if the excludes were completely ignored, this would still pass :(

In order to check that these files are actually being excluded, we'd need to either manually check the `build.ninja`, or potentially do something in `build_tests.sh`: e.g. `touch an.implicit.src bn.src`, then rerun the build and check that `gen_source_globbed_exclude_implicit_sources` hasn't been rebuilt.

We can do that in a follow-up commit though so I'll approve this review for now :)